torch/distributed/checkpoint/_state_dict_utils.py

```diff
@@ -49,13 +49,34 @@ def _all_gather_sharded_tensor(
 
 def _gather_state_dict(
     state_dict: Dict[str, Any],
+    *,
     pg: Optional[dist.ProcessGroup] = None,
     device: Optional[torch.device] = None,
+    cpu_offload: bool = False,
+    ranks_only: Tuple[int, ...] = tuple(),
 ) -> Dict[str, Any]:
     """
-    Given a state_dict, this API gathers all the ShardedTensors or DTensors in the state_dict.
+    Given a state_dict, this API gathers all the ShardedTensors or DTensors in
+    the state_dict.
+
+
+    Args:
+        state_dict (Dict[str, Any]): the target sharded state_dict.
+        pg (Optional[dist.ProcessGroup]): the process group that is used to
+            gather ShardedTensor.
+        device: (Optional[torch.device]): the device that is used to
+            perform allgather for ShardedTensor.
```

Comment:
nit: gather DTensor or ShardedTensor.

Reply:
ShardedTensor only. DTensor uses DeviceMesh to gather the tensors.